sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusStore.scala

```diff
@@ -146,4 +146,6 @@ class SparkPlanGraphNodeWrapper(
 case class SQLPlanMetric(
     name: String,
     accumulatorId: Long,
-    metricType: String)
+    metricType: String,
+    nodeId: Option[Long] = None,
```

Comment:
shall we use the node name and id in SparkPlanGraphNode instead?

Reply:
Current implementation uses `SQLExecutionUIData` and exposes `job` and `stage` details through it. However, `SQLExecutionUIData` does not have `SparkPlanGraphNode` data where it has `SQLPlanMetric`.

Reply:
Both `nodeId: Option[Long] = None` and `nodeName: Option[String] = None` were removed from `SQLPlanMetric` and provided by `sqlStore.planGraph(executionId)` instead.